src/systems/spaceship_collisions.rs

```diff
@@ -114,11 +114,11 @@ impl<'s> System<'s> for SpaceshipBlastCollisionSystem {
         for event in collision_event_channel.read(self.event_reader.as_mut().unwrap()) {
             // Is the player colliding with an entity with a blast component?
             if let Some(blast) = blasts.get_mut(event.colliding_entity) {
-                let spaceship = spaceships.get_mut(event.player_entity).unwrap();
+                let spaceship_health = healths.get_mut(event.player_entity).unwrap();
                 let blast_transform = transforms.get(event.colliding_entity).unwrap();
 
                 //first check if the blast is allied with the player
-                //if the blast collides with the player and the player is not currently barrel rolling the blast hits
+                //TODO blast should not hit if player is currently barrel rolling
```

Comment:
we should probably file a bug for this. 

also nit: put a whitespace here 
```suggestion
                // TODO blast should not hit if player is currently barrel rolling
```